Overlay menu new header etc

axelra82 opened the pull request:
Put in an overlay menu, some re-structuring (see new JS file + moved from head to end of body). Static Covid 19 top banner for header (i.e. on all pages), "student-info-banner" on index started.

Rearranged header and made alt. version for logotype that suits this layout better.

axelra82 committed:
Remove log output on componentLoader

axelra82 committed:
Add lock class to body to prevent scroll on menu overlay

axelra82 committed:
Give lock class in reset less general (conflict risk) class name, scroll-lock.

axelra82 committed:
Add general hover color variable in colors

axelra82 committed:
Move component loader to body end and add inital menu toggle test

axelra82 committed:
Inital restructure of header for new menu (overlay toggle)

axelra82 committed:
New header styles to accommodate new overlay menu toggle function

axelra82 committed:
Start working on overlay toggle menu

axelra82 committed:
Init componentLoader in js file and move script load in HTML to body end, before menu toggle loads

axelra82 committed:
Initial ovelay menu test as per group discussion. Minimal testing done but should lock body scroll on open and work on all sizes

axelra82 committed:
Add new color for background on covid-19 top banner

axelra82 committed:
Add background color for index page "student-banner"

axelra82 committed:
Add covid-19 top full width banner

axelra82 committed:
Add student-banner on index

axelra82 committed:
.gitignore fix

axelra82 committed:
Absolute root file referencing

axelra82 committed:
.

axelra82 committed:
Absolute reference for logotype source in header

axelra82 committed:
Update heading for loop in typography to include 1-6 (prev. 1-3)

axelra82 committed:
Start touch up on student info banner

axelra82 committed:
Ref fix for files, going back to root ref.

axelra82 committed:
Reset/clear webkit aparance for search input

axelra82 committed:
Initial testing on search function done

axelra82 committed:
Change component loader name to match file naming convention

axelra82 committed:
Streamline component loader function

stenwall reviewed (changes requested):
These two files are conflicting with main branch:
> index.html
> js/componentLoader.js

I can click *Resolve conflicts* but since it's js and stuff I'm not sure how to fix it. I can't merge until the conflicts are resolved. Can you look at the files @axelra82?

axelra82 committed:
Enclose search elements in new search box element and remove script tag (all scripts will load in component-loader init)

axelra82 committed:
Clean up search function and adapt to new elements. Better structure.

axelra82 committed:
Fix searching styling (while search is performed)

axelra82 committed:
Update page publikationer to reflect new script loading procedure

axelra82 committed:
Done testing header, remove hard coded header and update script loading procedure

axelra82 committed:
Component loader update. For live testing

axelra82 committed:
New processed SCSS

axelra82 committed:
GitHub pages in organisation makes thing quirky

For testing I changed the file references to match repo name 'u02-redesign-unit7', in online editor.

axelra82 committed:
Same thing as previous. Have to manually update file references

axelra82 committed:
Manual changes for testing

All file references should be absolute to root, not relative.

axelra82 committed:
Quick online debug

axelra82 committed:
Test async attribute on script at body end

axelra82 committed:
Merge branch 'overlay-menu-new-header-etc' of github.com:chas-academy/u02-redesign-unit7 into overlay-menu-new-header-etc

axelra82 committed:
Fix merge conflicts

axelra82 committed:
Try script async in head

axelra82 committed:
Fix file reference

axelra82 committed:
Test script defer

axelra82 committed:
Github pages location test

axelra82 committed:
Remove log in component loader and bring back var !null check for event listener vars

axelra82 committed:
Add commenting for index head script

axelra82 commented:
I made to many changes during the time the request was up. Closing request and merging locally to start new branch for scroll to top function 👍
